Source/Core/DolphinQt/CMakeLists.txt

```diff
@@ -1,15 +1,24 @@
-if (NOT Qt5_DIR AND MSVC)
+if (Qt6_DIR)
+	set(Qt_DIR "${Qt6_DIR}")
+endif()
+
+if (Qt5_DIR)
+	set(Qt_DIR "${Qt5_DIR}")
+endif()
```

Comment:
I could be failing to read CMake properly, but it seems like if Qt5_DIR and Qt6_DIR are both set it'll use Qt5. Is that intended?

Reply:
i think this section is for manually overriding the Qt installation directory, so you assume the user won't call cmake with both directories defined. The auto-finding of the Qt installation should be done by the following `find_package(Qt NAMES Qt6 Qt5...` call, I assume that chooses higher version as expected but haven't tested.

Reply:
so, you're correct in that all this version detection logic was not working correctly. i'll push what i got working in a few. strangely the documented method doesn't seem to actually work.